### ISSUE TITLE
fence_vmware_soap plugin for Ubuntu 16.04

### ISSUE THREAD

**Issue by elwood218** [ClusterLabs/pcs#168]
Hi there,

I am running Ubuntu 16.04 and cannot use "fence_vmware_soap". The Binary is there but the Plugin is missing.

```
pcs stonith list | grep Error
Error: no metadata for /usr/sbin/fence_compute
Error: no metadata for /usr/sbin/fence_ovh
Error: no metadata for /usr/sbin/fence_vmware_soap
```

I was trying to add plugin out of Github by copying it to `/usr/lib/stonith/plugins/external/fence_vmware_soap` and it is successfully shown by `stonith -L`.

BUT `pcs -f cluster_pgsql stonith create fence_vm_foo external/fence_vmware_soap ...` is still throwing following error:
`"Error: Unable to create resource 'stonith:external/fence_vmware_soap', it is not installed on this system (use --force to override)"`

**Comment by tomjelinek** [comment]
Hi @elwood218,

`Error: no metadata for /usr/sbin/fence_vmware_soap` means the agent does not provide valid metadata and therefore cannot be used by pcs. Output of `/usr/sbin/fence_vmware_soap -o metadata` should provide some hints what's wrong with the agent. Can you post it?

If I'm not mistaken, Ubuntu 16.04 ships pcs-0.9.149. This version only supports fence agents in `/usr/sbin/`. For `external/fence_*` agents you need at least pcs-0.9.155. This issue has been covered at https://github.com/ClusterLabs/pcs/issues/81.

**Comment by elwood218** [comment]
Hi tomjelinek :)
Thanks for answering that fast! Your comment helped already - this is nice to know. :)

```
/usr/sbin/fence_vmware_soap -o metadata
Traceback (most recent call last):
  File "/usr/sbin/fence_vmware_soap", line 4, in <module>
    import shutil, tempfile, suds
ImportError: No module named suds
```

So I will install suds. I guess this will solve the problem and I dont need any external agent. If not I will take a look at issue #81 .
I will let you know if it is solved.

**Comment by elwood218** [comment]
This fixed the problem! Thanks!

So I have installed:
apt-get install python-suds
apt-get install python-requests

The stonith resource was created then by pcs and the metadata was available with `pcs stonith list`.

Now I have another problem with the `/usr/sbin/fence_vmware_soap` itself.
We already found out that obviusly `--ssl-insecure` is set in the script but not really used. The difference to the script which is working on CentOS is following:
```
45,49d44
< 			import ssl
< 			from requests.packages.urllib3.exceptions import InsecureRequestWarning
< 			if hasattr(ssl, '_create_unverified_context'):
< 				ssl._create_default_https_context = ssl._create_unverified_context
< 			requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
```

As this is not related to pcs or in general you guys. I will try to find solution somewhere else - but just wanted to put it here - because maybe someone also fenced that problem already.
Will close this now anyway.. :)
